university-agent/src/main/java/nl/quintor/studybits/service/AgentService.java

```diff
@@ -23,46 +20,41 @@
 import java.io.IOException;
 import java.io.UnsupportedEncodingException;
 import java.util.*;
-import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.ExecutionException;
 
 import static nl.quintor.studybits.indy.wrapper.message.IndyMessageTypes.*;
 
 @Service
 @Slf4j
 public class AgentService {
-
     @Autowired
     private TrustAnchor universityTrustAnchor;
-
     @Autowired
     private Issuer universityIssuer;
-
     @Autowired
     private Verifier universityVerifier;
-
     @Autowired
     private IdentityService identityService;
-
     @Autowired
     private StudentService studentService;
-
     @Autowired
     private CredentialDefinitionService credentialDefinitionService;
-
     @Autowired
     private ExchangePositionService exchangePositionService;
+    @Autowired
+    private MessageEnvelopeCodec messageEnvelopeCodec;
 
     @Value("${nl.quintor.studybits.university.name}")
     private String universityName;
 
+
+
     public MessageEnvelope processMessage(MessageEnvelope messageEnvelope) throws IndyException, ExecutionException, InterruptedException, IOException {
-        String messageTypeURN = messageEnvelope.getType().getURN();
+        String messageTypeURN = messageEnvelope.getMessageType().getURN();
 
-        messageEnvelope.setIndyWallet(universityIssuer);
 
         if (messageTypeURN.equals(CONNECTION_RESPONSE.getURN())) {
-            return handleConnectionResponse(messageEnvelope);
+            return handleConnectionResponse(MessageEnvelope.convertEnvelope(messageEnvelope, CONNECTION_RESPONSE));
```

Comment:
Shouldn't you convert the envelop on CREDENTIAL_REQUEST and PROOF as well?

Reply:
Done